packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -329,6 +349,7 @@ export class MediaPlaceholder extends Component {
 									</IconButton>
 									{ mediaLibraryButton }
 									{ this.renderUrlSelectionUI() }
+									{ this.renderCancelLink() }
```

Comment:
Minor: If we find ourselves creating "render" functions, it serves as a minor code smell that it ought to be either (a) a formal component to encapsulate the complex logic or (b) rendered inline. In how it's used here, we have a bigger problem that our `render` function creates entirely different element trees on some conditions. There's room for refactoring here to create common assignments of elements to variables and reuse them across conditions. Fair to say that's outside the scope of this effort.

Reply:
Worth looking back at the history of MediaPlaceholder as well. It seems as though it was previously mostly rendered inline and then refactored into methods.

Reply:
Indeed, I "copied" the way `renderUrlSelectionUI ` removed and reused part of the code, b/c I am not react-confident enough to have opinions about code organisation in react components :) I will make a note and discuss this separately though and resolve this conversation to allow this PR to move on.